packages/store/addon/-private/system/ds-model-store.ts

```diff
@@ -156,15 +157,18 @@ class Store extends CoreStore {
     let maybeFactory = this._modelFactoryFor(modelName);
 
     // for factorFor factory/class split
-    let klass = maybeFactory.class ? maybeFactory.class : maybeFactory;
-    if (!klass.isModel) {
-      return new ShimModelClass(this, modelName);
+    let klass = maybeFactory && maybeFactory.class ? maybeFactory.class : maybeFactory;
+    if (!klass || !klass.isModel) {
+      if (!CUSTOM_MODEL_CLASS || !this.getSchemaDefinitionService().doesTypeExist(modelName)) {
```

Comment:
Should this be in the prod build?

Reply:
Yes? Why wouldn’t it be?

Reply:
Are you asking if we should reconsider the existence of this prod validation? (It is currently there in prod but I'd be willing to have it be stripped)